Changelog.md

```diff
@@ -6,6 +6,8 @@ Language Features:
 Compiler Features:
  * NatSpec: Add fields "kind" and "version" to the JSON output.
  * Commandline Interface: Prevent some incompatible commandline options from being used together.
+ * Yul Optimizer: LoopInvariantCodeMotion can move `extcodehash`, `extcodesize`, `sload`, `returndatasize`, `balance` and  `selfbalance`, if the loop does not invalidate the storage.
```

Comment:
Not sure if we have the framework already, but it would be nice to have a test that compiles solidity code that creates the sum of a storage array and compiles via yul. We would have an `sload` for the array size for the loop bound check and for index access (to check for out-of-bounds array access). The optimizer should now be able to pull the `sload` out of the loop. In addition to that, it might even optimize away the bounds check (but I'm not sure about that yet).

So the test would generate optimized ir code - maybe a commandline test?